### PR TITLE
Move all benchmark circuit generation to `mitiq.benchmarks`, adding option for conversions

### LINKED ISSUE #589: Move all benchmark circuits to mitiq.benchmarks
Discussed at Mitiq meeting. All benchmark circuits should be defined in `mitiq.benchmarks` and have the option to output as a specific supported type (i.e., `QPROGRAM`). This is preferable to (re)defining them in `mitiq_qiskit`, `mitiq_pyquil`, etc.

### PR DESCRIPTION
**Pull request by Aaron-Robertson**
Description
-----------

Resolves #589.

Removes benchmarking circuits for testing and moves conversion to `mitiq.benchmarks.generate_rb_circuits()` with supported type outputs.

### CODE REVIEW

**Comment by Aaron-Robertson** [comment]
@rmlarose I don't think this one's ready for a full review, but could you take a look and give some conceptual/broad stroke feedback?

It's a working solution based on testing. I tried to take into account the comments I saw on the original issue, and the added unit tests are sort of dummy tests for now (maybe even until the generalized executors).

**Comment by rmlarose** [comment]
> @rmlarose I don't think this one's ready for a full review, but could you take a look and give some conceptual/broad stroke feedback?

Thanks @Aaron-Robertson, looks generally good. 

- I would just add `return_type` to `generate_rb_circuits` instead of creating a new function. I think that's preferable unless you had a good reason for another function?
- I wouldn't include `random_one_qubit_identity_circuit` as its own function. If this was a utility for testing somewhere it can be defined there, but it doesn't need to be in the public API.
  - Actually, I would be in favor of removing this function altogether, even from tests. It's simpler and fewer lines of code to just call `generate_rb_circuits` with `num_qubits=1`. 
- Conversions can be slow, so I:
  - Would conservatively test with `return_type` as `QPROGRAM \ Cirq`, e.g. a few circuits with a small `num_cliffords`.
  - Only execute when `return_type` is Cirq. We don't need to test executing the other circuits here, just that they are returned as the right type.

**Comment by Aaron-Robertson** [comment]
Sounds good and can do! With conversion built in, I agree it makes sense to remove `random_one_qubit_identity_circuit` post proof of concept.

I included the separate function for `converted_rb_circuits` to try to answer a concern I saw on #320 that 
> Benchmarks is for testing mitiq in its native representation

But I'm happy to combine them.

**Comment by Aaron-Robertson** [comment]
@rmlarose This one's ready for review if you have a chance to review/request reviews.

File: mitiq/benchmarks/randomized_benchmarking.py [hunk 1]
```diff
@@ -62,6 +79,19 @@ def generate_rb_circuits(
...
+                _random_two_q_clifford(
+                    *qubits,  # type: ignore
+                    num_cliffords,
+                    cfd_matrices,
+                    cliffords,
+                ),
+                return_type,
+            )
+            for _ in range(trials)
+        ]
```
Comment (rmlarose) [inline mitiq/benchmarks/randomized_benchmarking.py:94]: I think it's (marginally) better to do something like below for readability / fewer LOC.

```python
circuits = [
        _random_two_q_clifford(
            *qubits,  # type: ignore
            num_cliffords,
            cfd_matrices,
            cliffords,
        )
        for _ in range(trials)
    ]

if return_type is None or return_type == "cirq":
    return circuits

return [convert_from_mitiq(circuit, return_type) for circuit in circuits]
```

File: mitiq/mitiq_pyquil/tests/test_zne_mitiq_pyquil.py [hunk 2]
```diff
@@ -30,28 +30,10 @@
     generate_qcs_executor,
     ground_state_expectation,
 )
-from mitiq.mitiq_pyquil.conversions import to_pyquil
 from mitiq.benchmarks.randomized_benchmarking import generate_rb_circuits
 
 TEST_DEPTH = 30
 
-
-def random_one_qubit_identity_circuit(num_cliffords: int) -> pyquil.Program:
```
Comment (rmlarose) [inline mitiq/mitiq_pyquil/tests/test_zne_mitiq_pyquil.py:39]: So much better without this everywhere 🙏

File: mitiq/benchmarks/randomized_benchmarking.py [hunk 3]
```diff
@@ -37,6 +42,8 @@ def generate_rb_circuits(
         num_cliffords: The number of Clifford group elements in the
             random circuits. This is proportional to the depth per circuit.
         trials: The number of random circuits at each num_cfd.
+        return_type: String specifier for Mitiq-supported circuit type
+            conversion.
```
Comment (rmlarose) [inline mitiq/benchmarks/randomized_benchmarking.py:46]:
```suggestion
        return_type: String which specifies the type of the returned circuits. See the keys of ``mitiq.QPROGRAM`` for options. If ``None``, the returned circuits are ``cirq.Circuit``s.
```

**Review by rmlarose — CHANGES_REQUESTED** [review]
Two small suggestions then LGTM.

File: mitiq/benchmarks/randomized_benchmarking.py (hunk 1 above)
Comment (Aaron-Robertson) [reply mitiq/benchmarks/randomized_benchmarking.py:94]: It's much better 😅  Can define the `random_single_q`s as `circuits` too then there's only one conditional return and one call to convert 💯

**Review by rmlarose — APPROVED** [review]

**Comment by rmlarose** [comment]
Thanks @Aaron-Robertson!

**Pull request merged**